staging/src/k8s.io/kubectl/pkg/util/term/term_writer.go

```diff
@@ -74,6 +74,28 @@ func NewWordWrapWriter(w io.Writer, limit uint) io.Writer {
 	}
 }
 
+func GetWordWrapperLimit() uint {
+	stdout := os.Stdout
+	fd := stdout.Fd()
+	if !term.IsTerminal(fd) {
+		panic("file descriptor is not a terminal")
```

Comment:
We should be able to return an error from the calling function.

https://pkg.go.dev/text/template@go1.17.1#FuncMap

Reply:
Yes we can but in the `flagUsages`, it's returning a string to the `template` so I wasn't sure returning an error is a good idea
```go
func flagsUsages(f *flag.FlagSet) string {
	flagBuf := new(bytes.Buffer)
	printer := NewHelpFlagPrinter(flagBuf)

	wrapLimit := term.GetWordWrapperLimit()
	minFlagLen := uint(getMinFlagLen(f))
	maxFlagLen := uint(getMaxFlagLen(f))
	tabWidth := maxFlagLen - minFlagLen // the maximum width that will be padded by tabWriter

	f.VisitAll(func(flag *flag.Flag) {
		if flag.Hidden {
			return
		}
		printer.PrintHelpFlags(flag, wrapLimit, tabWidth, minFlagLen, maxFlagLen)
	})
	printer.FlushTabWriter()
	return flagBuf.String()
}
```

Reply:
I was thinking since `template.FuncMap` functions can return an error that you could pass it from `flagsUsages` as well. Not sure what the behavior is though for errors there.

Reply:
`template.FuncMap` is intended to replace the input and output a string for the [`template`](https://github.com/kubernetes/kubectl/blob/159a770147fb28337c6807abb1b2b9db843d0aff/pkg/util/templates/templates.go#L24-L70)

Reply:
I took a quick look at this and returning an error from those can be handled gracefully [here](https://github.com/kubernetes/kubernetes/blob/cfe3155d35d034d44c44c1699a85aae3ac6e15d1/staging/src/k8s.io/kubectl/pkg/util/templates/templater.go#L94).

```
_output/bin/kubectl get --help 1>/dev/null
Error: template: usage:12:43: executing "usage" at <flagsUsages $visibleFlags>: error calling flagsUsages: file descriptor is not a terminal
```

```diff
diff --git a/staging/src/k8s.io/kubectl/pkg/util/templates/templater.go b/staging/src/k8s.io/kubectl/pkg/util/templates/templater.go
index d2d5e6d87ed..6914d0cda4f 100644
--- a/staging/src/k8s.io/kubectl/pkg/util/templates/templater.go
+++ b/staging/src/k8s.io/kubectl/pkg/util/templates/templater.go
@@ -218,11 +218,14 @@ func (t *templater) usageLine(c *cobra.Command) string {
 	return usage
 }
 
-func flagsUsages(f *flag.FlagSet) string {
+func flagsUsages(f *flag.FlagSet) (string, error) {
 	flagBuf := new(bytes.Buffer)
 	printer := NewHelpFlagPrinter(flagBuf)
 
-	wrapLimit := term.GetWordWrapperLimit()
+	wrapLimit, err := term.GetWordWrapperLimit()
+	if err != nil {
+		return "", err
+	}
 	minFlagLen := uint(getMinFlagLen(f))
 	maxFlagLen := uint(getMaxFlagLen(f))
 	tabWidth := maxFlagLen - minFlagLen // the maximum width that will be padded by tabWriter
@@ -234,7 +237,7 @@ func flagsUsages(f *flag.FlagSet) string {
 		printer.PrintHelpFlags(flag, wrapLimit, tabWidth, minFlagLen, maxFlagLen)
 	})
 	printer.FlushTabWriter()
-	return flagBuf.String()
+	return flagBuf.String(), nil
 }
 
 func getFlagFormatWithTab(f *flag.Flag) string {
diff --git a/staging/src/k8s.io/kubectl/pkg/util/term/term_writer.go b/staging/src/k8s.io/kubectl/pkg/util/term/term_writer.go
index 26d429ad033..17d59e25832 100644
--- a/staging/src/k8s.io/kubectl/pkg/util/term/term_writer.go
+++ b/staging/src/k8s.io/kubectl/pkg/util/term/term_writer.go
@@ -17,6 +17,7 @@ limitations under the License.
 package term
 
 import (
+	"errors"
 	"io"
 	"os"
 
@@ -74,15 +75,15 @@ func NewWordWrapWriter(w io.Writer, limit uint) io.Writer {
 	}
 }
 
-func GetWordWrapperLimit() uint {
+func GetWordWrapperLimit() (uint, error) {
 	stdout := os.Stdout
 	fd := stdout.Fd()
 	if !term.IsTerminal(fd) {
-		panic("file descriptor is not a terminal")
+		return 0, errors.New("file descriptor is not a terminal")
 	}
 	terminalSize := GetSize(fd)
 	if terminalSize == nil {
-		panic("terminal size is nil")
+		return 0, errors.New("terminal size is nil")
 	}
 	var limit uint
 	switch {
@@ -93,7 +94,7 @@ func GetWordWrapperLimit() uint {
 	case terminalSize.Width >= 80:
 		limit = 80
 	}
-	return limit
+	return limit, nil
 }
 
 func (w wordWrapWriter) Write(p []byte) (nn int, err error) {
```